lib/checks/aria/required-parent.js

```diff
@@ -1,87 +1,41 @@
-function getSelector(role) {
-	var impliedNative = axe.commons.aria.implicitNodes(role) || [];
-	return impliedNative.concat('[role="' + role + '"]').join(',');
+const { aria } = axe.commons;
+
+const role = aria.getRole(virtualNode);
+const requiredContext = aria.requiredContext(role);
+if (!requiredContext) {
+	return true;
 }
 
-function getMissingContext(virtualNode, requiredContext, includeElement) {
-	var index,
-		length,
-		role = virtualNode.actualNode.getAttribute('role'),
-		missing = [];
+const owner = aria.getOwnerVirtual(virtualNode);
+if (!owner) {
+	return undefined;
+}
+if (hasRequiredContextRole(owner, requiredContext)) {
+	return true;
+}
 
-	if (!requiredContext) {
-		requiredContext = axe.commons.aria.requiredContext(role);
-	}
+this.relatedNodes(owner);
+this.data(requiredContext);
+return false;
 
-	if (!requiredContext) {
-		return null;
+/**
+ * Verify if a given element or its parent (recursively) has any of the required context roles
+ * @param {HTMLElement} elm element
+ * @param {String[]} contextRoles list of context roles
+ * @returns {Boolean}
+ */
+function hasRequiredContextRole(elm, contextRoles) {
+	if (!elm) {
+		return false;
 	}
 
-	for (index = 0, length = requiredContext.length; index < length; index++) {
-		if (
-			includeElement &&
-			axe.utils.matchesSelector(
-				virtualNode.actualNode,
-				getSelector(requiredContext[index])
-			)
-		) {
-			return null;
-		}
-		if (
-			axe.commons.dom.findUpVirtual(
-				virtualNode,
-				getSelector(requiredContext[index])
-			)
-		) {
-			//if one matches, it passes
-			return null;
-		} else {
-			missing.push(requiredContext[index]);
-		}
+	const role = aria.getRole(elm);
+	if (['presentation', 'none', null].includes(aria.getRole(elm))) {
```

Comment:
You've already tested for presentation/none in getOwnerVirtual, why are you doing this again? Also, why are you ignoring null? Firefox doesn't ignore null roles, so I don't think we should either. It'd be good to add an option to this check to make it configurable whether or not null is ignored, but that should probably be handled by getOwnerVirtual. Can you either add that or open an issue for it?